### PR TITLE
build: update sonic & golang version in ci

### LINKED ISSUE #626: Optimize: add build tag to limit the available Golang version of sonic deps
**Is your feature request related to a problem? Please describe.**

When the middle version of Golang is released and sonic is incompatible, if our users directly upgrade Golang, the compilation will fail directly. 

This issue is not very friendly for our users.

**Describe the solution you'd like**

Perhaps adding some build limitation for sonic deps - limit the available Golang version - fallback to standard JSON lib if Golang version from users' env is beyond this limitation.

**Describe alternatives you've considered**

**Additional context**

Related issues: #579 #212

### PR DESCRIPTION
**Pull request by wzekin**
#### What type of PR is this?

chore
#### Check the PR title.

- [ ] This PR title match the format: \<type\>(optional scope): \<description\>
- [ ] The description of this PR title is user-oriented and clear enough for others to understand.

#### (Optional) Translate the PR title into Chinese.
更新 sonic 以及 ci 中 golang 的版本

#### (Optional) More detail description for this PR(en: English/zh: Chinese).

en: update sonic & golang version in ci
zh(optional): 更新 sonic 以及 ci 中 golang 的版本

#### Which issue(s) this PR fixes:
Sonic v1.8.0 begin to support #626 (closes #626 )

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ jobs:
   lint-and-ut:
     strategy:
       matrix:
-        version: [ 1.18, 1.19 ]
+        version: [ '1.19', '1.20' ]
```
Comment (welkeyever) [inline .github/workflows/tests.yml:9]: 能多加几个版本一起测测吗

File: go.sum [hunk 2]
```diff
@@ -4,8 +4,12 @@ github.com/bytedance/gopkg v0.0.0-20220413063733-65bf48ffb3a7 h1:PtwsQyQJGxf8iaP
 github.com/bytedance/gopkg v0.0.0-20220413063733-65bf48ffb3a7/go.mod h1:2ZlV9BaUH4+NXIBF0aMdKKAnHTzqH+iMU4KUjAbL23Q=
 github.com/bytedance/sonic v1.5.0 h1:XWdTi8bwPgxIML+eNV1IwNuTROK6EUrQ65ey8yd6fRQ=
 github.com/bytedance/sonic v1.5.0/go.mod h1:ED5hyg4y6t3/9Ku1R6dU/4KyJ48DZ4jPhfY1O2AihPM=
```
Comment (welkeyever) [inline go.sum:6]: go mod verify 一下

File: .github/workflows/tests.yml (hunk 1 above)
Comment (wzekin) [reply .github/workflows/tests.yml:9]: 上几个版本，再加一个 1.18？

File: go.sum (hunk 2 above)
Comment (wzekin) [reply go.sum:5]: done

File: .github/workflows/tests.yml (hunk 1 above)
Comment (welkeyever) [reply .github/workflows/tests.yml:9]: 可以的

**Review by Duslia — APPROVED** [review]

**Review by FGYFFFF — APPROVED** [review]

**Pull request merged**